pkg/apis/infra/v1alpha1/cluster_types.go

```diff
@@ -226,13 +226,27 @@ type ResourceRef struct {
 type ClusterStatus struct {
 	// Conditions defines current service state of the cluster.
 	// +optional
-	Conditions []capiv1.Conditions `json:"conditions,omitempty"`
+	Conditions capiv1.Conditions `json:"conditions,omitempty"`
 	// Phase is the current lifecycle phase of the cluster.
+	// +optional
 	Phase string `json:"phase,omitempty"`
 	// APIEndpoint is the endpoint to communicate with the apiserver.
+	// +optional
 	APIEndpoint string `json:"apiEndpoint,omitempty"`
 	// ServiceAccountIssuer is the URL of the service account issuer.
+	// +optional
 	ServiceAccountIssuer string `json:"serviceAccountIssuer"`
+	// ObservedGeneration is the most recent generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
what does this mean?

Reply:
https://alenkacz.medium.com/kubernetes-operator-best-practices-implementing-observedgeneration-250728868792

Reply:
It is still ambiguous while looking at the link to Cluster API. Actually in that PR no controller is setting up `ObservedGeneration`

Reply:
And what could it show to users or controllers